packages/shopify-api/lib/auth/oauth/create-session.ts

```diff
@@ -0,0 +1,70 @@
+import {v4 as uuidv4} from 'uuid';
+
+import {Session} from '../../session/session';
+import {ConfigInterface} from '../../base-types';
+import {RequestReturn} from '../../clients/http_client/types';
+import {logger} from '../../logger';
+import {getJwtSessionId, getOfflineId} from '../../session/session-utils';
+
+import {
+  AccessTokenResponse,
+  OnlineAccessResponse,
+  OnlineAccessInfo,
+} from './types';
+
+export function createSession({
+  config,
+  postResponse,
+  shop,
+  state,
+}: {
+  config: ConfigInterface;
+  postResponse: RequestReturn;
```

Comment:
Should this function take in the parsed data so it doesn't assume all POSTs are going to return the same values forever? Or is that something we can guarantee won't change from Shopify's side?

Reply:
that's a good point. I've added a function that'll convert the response to the appropriate type.